dev.py

```diff
@@ -1174,26 +1176,30 @@ def task_check_unicode():
 def task_check_test_name():
     # emit_cmdstr(os.path.join('tools', 'check_test_name.py'))
     return {
-        "basename": "check-testname",
+        "basename": "check_testname",
         "actions": [str(Dirs().root / "tools" / "check_test_name.py")],
         "doc": "Check tests are correctly named so that pytest runs them."
     }
 
 
 @cli.cls_cmd('lint')
 class Lint:
-    """:dash: Run linter on modified files and check for
+    """:dash: Run linter on modified (or all) files and check for
     disallowed Unicode characters and possibly-invalid test names."""
     fix = Option(
         ['--fix'], default=False, is_flag=True, help='Attempt to auto-fix errors'
     )
+    all = Option(
+        ['--all'], default=False, is_flag=True,
+        help='lint all files instead of just modified files.'
+    )
 
     @classmethod
-    def run(cls, fix):
+    def run(cls, fix, all):
         run_doit_task({
-            'lint': {'fix': fix},
-            'check-unicode': {},
-            'check-testname': {},
+            'lint': {'fix': fix, 'all': all},
+            'check_unicode': {},
```

Comment:
this fixes a small bug from gh-22003

Reply:
Can we make all of these use dash or all use underscore?
Changing `check-unicode` to `unicode_check` when they are sitting next to `check-testname` and `check_python_h_first` is confusing. :}
Not a big deal... just would help me remember better...

Reply:
Sorry, I'm not sure what you mean? I think everything changed here in that regard is just implementation details?

Reply:
I see.... people can't do `dev.py check_unicode`.  I was thinking they were developer facing.
Sorry for the noise.

Reply:
no worries - maybe we should add that! Although the whole lint check is fast enough that I'm not sure it matters much.